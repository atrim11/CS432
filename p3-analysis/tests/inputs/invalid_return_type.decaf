def int main() {

    funcCallA();
    funcCallB();
    funcCallC();
    funcCallD();
    return 9999999;
}

def void funcCallA() {
    return -1;
}

def int funcCallB() {
    return true;
}

def int funcCallE() {
}


def bool funcCallC() {
    return 0xfeedface;
    return 0xdecaf;
}

def void funcCallD(int a, int b, int c, int d, int e, int f) {
    return 0xdeadbeef;
}