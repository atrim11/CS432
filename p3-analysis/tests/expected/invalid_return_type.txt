Invalid number of function arguments on line 6
Invalid non-void return from void function on line 11
Type mismatch: int expected but bool found on line 15
Type mismatch: bool expected but int found on line 23
Type mismatch: bool expected but int found on line 24
Invalid non-void return from void function on line 28
